### PR TITLE
fix(admin): ensure superuser can always log in

### LINKED ISSUE #27: Initial Superuser can get in a "stuck" state
**Describe the bug**
If a super user is created but then - say, because they get distracted watching Galaxy of Terror 😅 - gets logged out without creating a team, they're stuck. They can't authenticate again that way because they aren't on a team, they never created one.

**To Reproduce**
1. Set up an empty portr instance
2. Log in and create the first superuser but DO NOT create a team
3. Delete cookie for session
4. Attempt to Log in again
5. See error message

**Expected behavior**
Superuser can log in even if they aren't on a team - so they can create a team

**Screenshots**

Error message is shown `User not part of any team`

**Additional context**
Fixed as part of #25 but there's a number of other changes there..

### PR DESCRIPTION
**Pull request by imnotjames**
when a superuser isn't on a team, they should still be able to log in so they can create a team -- this is a normal and expected flow for the superuser

fixes #27

### CODE REVIEW

File: admin/src/portr_admin/services/user.py [hunk 1]
```diff
@@ -14,6 +14,32 @@ class EmailFetchError(ServiceError):
     pass
 
 
+@transactions.atomic()
+async def get_or_create_user(email: str):
+    has_users = await User.filter().exists()
+
+    if not has_users:
+        # If this is the first user we can create them as a superuser
+        await User.create(
```
Comment (amalshaji) [inline admin/src/portr_admin/services/user.py:23]: return here

File: admin/src/portr_admin/tests/service_tests/test_user_service.py [hunk 2]
```diff
@@ -39,6 +39,69 @@ async def test_get_or_create_user_from_github_with_not_part_of_any_team(
...
+    async def test_get_or_create_user_from_github_without_email(
+        self, get_access_token_fn, get_user_fn
+    ):
+        get_access_token_fn.return_value = "token"
+        get_user_fn.return_value = {"email": "example@example.com"}
+
+        with pytest.raises(user_service.UserNotFoundError) as e:
+            await user_service.get_or_create_user_from_github("code")
+
+        assert str(e.value) == "User does not exist"
```
Comment (amalshaji) [inline admin/src/portr_admin/tests/service_tests/test_user_service.py:103]: Duplicate test case

**Review by amalshaji — CHANGES_REQUESTED** [review]

File: admin/src/portr_admin/services/user.py [hunk 3]
```diff
@@ -35,16 +61,7 @@ async def get_or_create_user_from_github(code: str):
...
-
-    if not is_superuser:
-        if not await TeamUser.filter(user__email=github_user["email"]).exists():
-            raise UserNotFoundError("User not part of any team")
-
-    user, _ = await User.get_or_create(
-        email=github_user["email"],
-        defaults={"is_superuser": is_superuser},
-    )
+    user = await get_or_create_user(github_user["email"])
```
Comment (amalshaji) [inline admin/src/portr_admin/services/user.py:64]: Now that I think of it, `get_or_create_user` should do exactly what it says. The check should be on the existing user, whether the user can log in or not.

**Review by amalshaji — CHANGES_REQUESTED** [review]

File: admin/src/portr_admin/services/user.py (hunk 3 above)
Comment (imnotjames) [reply admin/src/portr_admin/services/user.py:64]: Okay - sure, I can agree with that.  You mean that the business logic for handling "Is this user authenticated & can log in" vs "I want to get a user for this email whether or not it exists" should be separate.

However, that would require changing things a bit more than I had here and would be closer to a refactor of the logic, right?

I can approach that, but I was trying to change as little as possible to fix the bug.

Comment (imnotjames) [reply admin/src/portr_admin/services/user.py:64]: Perhaps a separate function in the `user` service `can_log_in` which has the business logic separately?

Comment (imnotjames) [reply admin/src/portr_admin/services/user.py:64]: Making that change and moving that check outside of both `get_or_create_user_from_github` and this new function `get_or_create_user` would change up the logic:

A `GithubUser` will be created even if a user can't technically log in.  Maybe that's ok?

Comment (imnotjames) [reply admin/src/portr_admin/services/user.py:64]: Made that change, let me know what you think.

Comment (amalshaji) [reply admin/src/portr_admin/services/user.py:67]: > A GithubUser will be created even if a user can't technically log in. Maybe that's ok?

We don't want to create users if they are not added to any teams.

Comment (amalshaji) [reply admin/src/portr_admin/services/user.py:67]: So there should be two checks
- If the user doesn't exist, can the user be created?
- If the user exists, can the user log in?

Comment (imnotjames) [reply admin/src/portr_admin/services/user.py:67]: A user won't be created unless they're the first user.  A user that already exists (as in they were invited) but hasn't been associated with GitHub would be associated with GitHub.

This is an edge case for a user that was invited but then removed from the team before they logged in the first time.

It's a `GitHubUser` link being created in an edge case - not a `User`.  So what you described in your comment for desired business logic is the current behavior.

Comment (amalshaji) [reply admin/src/portr_admin/services/user.py:67]: Linking github user is fine

**Review by amalshaji — APPROVED** [review]
🎉

**Pull request merged**